include/numerics/petsc_solver_exception.h

```diff
@@ -39,12 +40,6 @@
 namespace libMesh
 {
 
-// Forward declarations
-namespace Parallel
-{
-class Communicator;
-}
-
```

Comment:
I don't see any other changes in this header that would require a full inclusion over a forward declaration. (I understand this PR might still be under development, so maybe there's another change I'm not seeing yet.)

Reply:
This is failing on devel due to my changes in the previous PRs which is why you probably don't see the changes that created the need for this change

Reply:
Probably should have separated this PR into two (still can if you would prefer)

Reply:
OK, it's up to you. I guess one reason to do it would be to make it easier to revert just the PetscFunctionBegin/Return stuff at some later time.

Reply:
Kk opened https://github.com/libMesh/libmesh/pull/3869